pinot-common/src/main/java/org/apache/pinot/common/function/scalar/regexp/RegexpLikeConstFunctions.java

```diff
@@ -34,17 +35,53 @@ public class RegexpLikeConstFunctions {
   @ScalarFunction
   public boolean regexpLike(String inputStr, String regexPatternStr) {
     if (_matcher == null) {
-      _matcher = PatternFactory.compile(regexPatternStr).matcher("");
+      Pattern p = PatternFactory.compile(regexPatternStr);
+      _matcher = p.matcher("");
```

Comment:
(minor) We can chain the calls to be more concise. Same for other calls and classes